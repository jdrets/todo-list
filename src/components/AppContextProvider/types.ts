import React from 'react'

export interface TasksTypes {
  status: string
  priority: string
  dueDate?: string
  title: string
  description: string
}

export interface AppContextProviderTypes {
  tasks: TasksTypes[]
  setTasks: React.SetStateAction<any>
  newTaskModal: boolean
  setNewTaskModal: React.SetStateAction<any>
}
