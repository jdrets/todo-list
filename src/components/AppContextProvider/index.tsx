import React, { FunctionComponent, useState, createContext, useEffect } from 'react'
import _ from 'lodash'

import TaskService from '../../services/taskService'
import { AppContextProviderTypes } from './types'

export const PageContext = createContext({} as AppContextProviderTypes)

const AppContextProvider: FunctionComponent<any> = ({ children, value = {} }) => {
  const [tasks, setTasks] = useState([])
  const [showCreateTaskModal, setShowCreateTaskModal] = useState(false)

  useEffect(() => {
    const storage = TaskService.get()
    setTasks(storage)
  }, [])

  const updateContextWithNewTask = (task) => {
    const newTasks = _.clone(tasks)
    newTasks.push(task)
    setTasks(_.reverse(newTasks))
  }

  const contextValues = {
    tasks,
    showCreateTaskModal,
    updateContextWithNewTask,
    setShowCreateTaskModal,
    ...value
  }

  return (
    <PageContext.Provider value={contextValues}>{children}</PageContext.Provider>
  )
}

export default AppContextProvider
