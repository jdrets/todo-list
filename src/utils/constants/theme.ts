const palette = {
  purpleDark: '#4B00BC',
  purple: '#6600FF',
  purpleMedium: '#8533FF',
  purpleLight: '#E3D0FF',
  green: '#59BC6F',
  greenLight: '#E6F5E2',
  red: '#EB5757',
  redLight: '#FCE4E4',
  yellow: '#DD9900',
  yellowLight: '#FEF1D8',
  white: '#FFFFFF',
  black: '#000000',
  gray: '#898989',
  grayLight: '#E0E0E0'
}

const theme = {
  colors: {
    primaryColor: palette.purple,
    primaryColorMedium: palette.purpleMedium,
    primaryColorLight: palette.purpleLight,
    primaryColorDark: palette.purpleDark,
    ...palette
  },
  layout: {
    breakpoints: {
      medium: '768px'
    },
    mobile: {
      width: '100%',
      padding: '0 24px'
    },
    desktop: {
      width: '1050px'
    }
  }
}

export default theme
