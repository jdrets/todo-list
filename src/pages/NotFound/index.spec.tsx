import React from 'react'
import { ThemeProvider } from 'styled-components'

import { fireEvent, render } from '@testing-library/react'

import NotFoundPage from '.'
import theme from '../../utils/constants/theme'
import AppContextProvider from '../../components/AppContextProvider'

const setup = () => (
  render(
    <ThemeProvider theme={theme}>
      <AppContextProvider>
        <NotFoundPage />
      </AppContextProvider>
    </ThemeProvider>
  )
)

const mockHistoryPush = jest.fn()

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: mockHistoryPush
  })
}))

describe('<NotFoundPage />', () => {
  test('should match with snapshot', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })

  test('should call push history function when button is clicked', () => {
    const { getByText } = setup()
    const button = getByText('Go to home')

    fireEvent.click(button)

    expect(mockHistoryPush).toHaveBeenCalled()
  })
})
