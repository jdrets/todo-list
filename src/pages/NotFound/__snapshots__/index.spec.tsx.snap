// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should match with snapshot 1`] = `
<div>
  <section
    class="sc-bdnylx jACAPs"
  >
    <section
      class="sc-gtssRu bHmfRF"
    >
      <div
        class="sc-dlnjPT cGRTAQ"
      >
        <div
          class="sc-eCApGN MUNNv"
        >
          <svg
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="m495.223 169.682h-75.35c.145-1.279.228-2.576.228-3.893 0-17.419-13.076-31.832-29.927-33.963v-34.986c0-10.052-8.178-18.23-18.23-18.23h-29.801c.589-2.393.907-4.891.907-7.463 0-17.231-14.019-31.25-31.25-31.25s-31.25 14.019-31.25 31.25c0 2.572.319 5.07.907 7.463h-146.263c-10.052 0-18.23 8.178-18.23 18.23v72.841h-21.922c-4.142 0-7.5 3.358-7.5 7.5s3.358 7.5 7.5 7.5h21.922v37.941h-101.964v-36.162c0-.98.797-1.778 1.777-1.778h41.238c4.142 0 7.5-3.358 7.5-7.5s-3.358-7.5-7.5-7.5h-41.238c-9.251 0-16.777 7.526-16.777 16.778v286.017c0 9.251 7.526 16.777 16.777 16.777h204.702c9.251 0 16.777-7.526 16.777-16.777v-26.611h35.486v26.611c0 9.251 7.526 16.777 16.777 16.777h61.051c4.142 0 7.5-3.358 7.5-7.5s-3.358-7.5-7.5-7.5h-61.051c-.98 0-1.777-.797-1.777-1.777v-26.611h83.201c10.052 0 18.23-8.178 18.23-18.229v-190.014h106.827v234.853c0 .98-.797 1.777-1.777 1.777h-106.625c-4.142 0-7.5 3.358-7.5 7.5s3.358 7.5 7.5 7.5h106.625c9.251 0 16.777-7.526 16.777-16.777v-286.016c0-9.252-7.526-16.778-16.777-16.778zm-90.122-3.893c0 10.612-8.633 19.246-19.245 19.246s-19.246-8.633-19.246-19.246c0-10.612 8.633-19.245 19.246-19.245 10.611 0 19.245 8.633 19.245 19.245zm-93.301-110.891c8.96 0 16.25 7.29 16.25 16.25s-7.29 16.25-16.25 16.25-16.25-7.29-16.25-16.25 7.289-16.25 16.25-16.25zm-176.606 38.712h154.914c5.625 5.434 13.272 8.787 21.692 8.787s16.067-3.353 21.692-8.787h38.452c1.781 0 3.23 1.449 3.23 3.23v36.418c-4.318 1.421-8.256 3.679-11.628 6.582h-231.582v-43c0-1.781 1.449-3.23 3.23-3.23zm236.75 337.255h-236.75c-1.781 0-3.23-1.449-3.23-3.229v-84.016c0-4.142-3.358-7.5-7.5-7.5s-7.5 3.358-7.5 7.5v84.016c0 10.052 8.178 18.229 18.23 18.229h88.063v26.611c0 .98-.797 1.777-1.777 1.777h-204.703c-.98 0-1.777-.797-1.777-1.777v-234.853h101.964v68.971c0 4.142 3.358 7.5 7.5 7.5s7.5-3.358 7.5-7.5v-151.754h221.451c-1.164 3.44-1.805 7.12-1.805 10.949 0 15.154 9.897 28.032 23.564 32.531v229.316c0 1.781-1.449 3.229-3.23 3.229zm18.23-208.242v-22.87c10.1-1.277 18.841-6.963 24.225-15.071h80.824c.98 0 1.777.797 1.777 1.778v36.164h-106.826z"
              />
              <path
                d="m275.491 302.898c-6.115-5.22-14.236-8.109-22.866-8.135-.039 0-.077 0-.116 0-8.586 0-16.68 2.838-22.798 7.999-3.167 2.67-3.569 7.402-.898 10.568 2.67 3.167 7.402 3.57 10.568.898 3.368-2.84 8.15-4.465 13.131-4.465h.067c5.006.015 9.807 1.671 13.173 4.544 1.413 1.206 3.143 1.795 4.866 1.795 2.119 0 4.225-.893 5.708-2.631 2.689-3.15 2.315-7.884-.835-10.573z"
              />
              <path
                d="m294.649 267.023c-.065-.002-.129-.011-.195-.011h-.024c-.002 0-.005 0-.007 0h-.057-.032c-.003 0-.007 0-.01 0-4.122 0-7.432 3.331-7.455 7.458-.022 3.897 2.973 7.115 6.778 7.501.056.006.114.005.17.01.184.015.367.03.555.031h.042c.258 0 .513-.013.765-.039 3.787-.379 6.745-3.575 6.745-7.462-.001-4.065-3.239-7.368-7.275-7.488z"
              />
              <path
                d="m212.844 267.012h-.088c-.014 0-.028 0-.042 0-4.122 0-7.432 3.331-7.455 7.458-.023 4.142 3.36 7.519 7.502 7.542h.042c4.122 0 7.476-3.331 7.499-7.458.024-4.143-3.316-7.519-7.458-7.542z"
              />
              <path
                d="m161.84 124.113c.013.001.025.004.038.006.164.017.332.021.499.028.089.004.177.014.267.015h.04c4.124 0 7.478-3.333 7.499-7.461.019-3.641-2.561-6.687-5.998-7.386-.009-.002-.019-.003-.029-.005-.231-.046-.466-.083-.704-.108-.085-.009-.173-.008-.259-.014-.157-.011-.313-.025-.472-.026h-.095c-.013 0-.026 0-.04 0-.258 0-.513.013-.764.039-3.782.379-6.697 3.575-6.697 7.462-.001 3.86 2.958 7.037 6.715 7.45z"
              />
              <path
                d="m185.359 124.162c4.142 0 7.5-3.358 7.5-7.5s-3.358-7.5-7.5-7.5h-.093c-4.142 0-7.454 3.358-7.454 7.5s3.405 7.5 7.547 7.5z"
              />
              <path
                d="m208.049 124.162c4.142 0 7.5-3.358 7.5-7.5s-3.358-7.5-7.5-7.5h-.093c-4.142 0-7.454 3.358-7.454 7.5s3.405 7.5 7.547 7.5z"
              />
              <path
                d="m55.094 140.562c1.464 1.464 3.384 2.197 5.303 2.197s3.839-.732 5.303-2.197l3.995-3.995 3.995 3.995c1.464 1.465 3.384 2.197 5.303 2.197s3.839-.732 5.303-2.197c2.929-2.929 2.929-7.677 0-10.606l-3.995-3.995 3.995-3.995c2.929-2.929 2.929-7.678 0-10.606-2.929-2.929-7.678-2.929-10.606 0l-3.995 3.995-3.995-3.995c-2.928-2.929-7.677-2.929-10.606 0s-2.929 7.677 0 10.606l3.995 3.995-3.995 3.995c-2.929 2.928-2.929 7.677 0 10.606z"
              />
              <path
                d="m430.557 131.263c1.464 1.464 3.384 2.197 5.303 2.197s3.839-.732 5.303-2.197l3.995-3.995 3.995 3.995c1.464 1.464 3.384 2.197 5.303 2.197s3.839-.732 5.303-2.197c2.929-2.929 2.929-7.678 0-10.606l-3.995-3.995 3.995-3.995c2.929-2.929 2.929-7.678 0-10.606-2.929-2.929-7.678-2.929-10.606 0l-3.995 3.995-3.995-3.995c-2.929-2.929-7.678-2.929-10.606 0-2.929 2.929-2.929 7.678 0 10.606l3.995 3.995-3.995 3.995c-2.93 2.928-2.93 7.677 0 10.606z"
              />
              <path
                d="m239.631 54.147c1.464 1.464 3.384 2.197 5.303 2.197s3.839-.732 5.303-2.197l3.995-3.995 3.995 3.995c1.464 1.464 3.384 2.197 5.303 2.197s3.839-.732 5.303-2.197c2.929-2.929 2.929-7.678 0-10.606l-3.995-3.995 3.995-3.995c2.929-2.929 2.929-7.678 0-10.606-2.929-2.929-7.678-2.929-10.606 0l-3.995 3.995-3.995-3.995c-2.929-2.929-7.678-2.929-10.606 0-2.929 2.929-2.929 7.678 0 10.606l3.995 3.995-3.995 3.995c-2.929 2.928-2.929 7.677 0 10.606z"
              />
            </g>
          </svg>
        </div>
        <div
          class="sc-hKFyIo dEftBs"
        >
          Oh, this page does not exist
        </div>
        <div
          class="sc-jSFkmK hsukcn"
        >
          Go to home
        </div>
      </div>
    </section>
  </section>
</div>
`;
